client/components/auth/auth.service.js

```diff
@@ -59,9 +59,7 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
     createUser(user, callback) {
       return User.save(user,
         function(data) {
-          $cookies.put('token', data.token);
-          currentUser = User.get();
-          return safeCb(callback)(null, user);
+          $location.path('/verify/emailVerification');
```

Comment:
We should not break the flow of the callback (and the overall signature consistency)- instead, we should add something in the user object to say it is unverified that the calling function can check and redirect